src/Repository/UserRepository.php

```diff
@@ -62,32 +76,52 @@ public function upgradePassword(PasswordAuthenticatedUserInterface $user, string
         $this->_em->flush();
     }
 
-    // /**
-    //  * @return User[] Returns an array of User objects
-    //  */
-    /*
-    public function findByExampleField($value)
+    public function setUserResetToken(string $emailAddress, Uuid $resetToken)
     {
-        return $this->createQueryBuilder('u')
-            ->andWhere('u.exampleField = :val')
-            ->setParameter('val', $value)
-            ->orderBy('u.id', 'ASC')
-            ->setMaxResults(10)
-            ->getQuery()
-            ->getResult()
-        ;
+        $forgottenUser = $this->findOneBy(['email' => $emailAddress]);
+
+        if (null !== $forgottenUser) {
+            $forgottenUser->setResetToken($resetToken);
+            $forgottenUser->setResetTokenCreatedAt(new \DateTime('now'));
+
+            $this->_em->persist($forgottenUser);
+            $this->_em->flush();
+        }
     }
-    */
 
-    /*
-    public function findOneBySomeField($value): ?User
+    public function validatingResetToken(Uuid $resetToken)
     {
-        return $this->createQueryBuilder('u')
-            ->andWhere('u.exampleField = :val')
-            ->setParameter('val', $value)
-            ->getQuery()
-            ->getOneOrNullResult()
-        ;
+        $forgottenUser = $this->findOneBy(['resetToken' => $resetToken]);
+        $userResetToken = $forgottenUser->getResetToken();
+
+        if ($userResetToken->compare($resetToken)) {
+            return 'Error';
+        }
+
+        $testTimestamp = $forgottenUser->getResetTokenCreatedAt()->getTimestamp();
+        $expiredTimestamp = $testTimestamp + 900;
+        $now = new \DateTime('now');
+        $nowTimestamp = $now->getTimestamp();
+
+        if ($nowTimestamp > $expiredTimestamp) {
+            return 'Expired Link';
+        }
+
+        return $forgottenUser;
+    }
+
+    public function setNewPassword(User $forgottenUser, string $plainPassword)
```

Comment:
A method called `setNewPassword` is a method that goes on the entity, not the repository.